corehq/apps/locations/models.py

```diff
@@ -957,14 +953,18 @@ def parent_id(self):
 
     @property
     def parent(self):
-        parent_id = self.parent_location_id
-        return Location.get(parent_id) if parent_id else None
+        if hasattr(self, '_sql_parent'):
+            return self._sql_parent.couch_location if self._sql_parent else None
```

Comment:
Yeah, for the latter case - if you instantiate it with parent, but haven't saved yet.